source/dub/internal/logging.d

```diff
@@ -380,3 +380,19 @@ string color(const string str, const Mode m = Mode.init)
 	else
 		return str;
 }
+
+private static int[16] tagWidthStack;
+private static int tagWidthIndex;
+
+void pushTagWidth(int w)
+{
+	tagWidthStack[tagWidthIndex] = tagWidth;
+	tagWidthIndex++;
+	assert(tagWidthIndex < tagWidthStack.length);
+	tagWidth = w;
+}
+
+void popTagWidth()
+{
+	tagWidth = tagWidthStack[--tagWidthIndex];
+}
```

Comment:
Move it into its own struct please, instead of 3 globals

Reply:
done